mkosi/__init__.py

```diff
@@ -429,18 +408,9 @@ def reset_machine_id(state: MkosiState) -> None:
         return
 
     with complete_step("Resetting machine ID"):
-        if not state.config.machine_id:
-            machine_id = state.root / "etc/machine-id"
-            machine_id.unlink(missing_ok=True)
-            machine_id.write_text("uninitialized\n")
-
-        dbus_machine_id = state.root / "var/lib/dbus/machine-id"
-        try:
-            dbus_machine_id.unlink()
-        except FileNotFoundError:
-            pass
-        else:
-            dbus_machine_id.symlink_to("../../../etc/machine-id")
+        machine_id = state.root / "etc/machine-id"
+        machine_id.unlink(missing_ok=True)
```

Comment:
if the rootfs is read-only, this should create an empty file instead - so that it can be used as a mount point for the ephemeral (or command-line driven) machine-id